### DISCUSSION TITLE
Should docs state PHP 8.1 or higher is required for v3.27.0 as internal dependencies seem to infer?

### DISCUSSION THREAD

**Discussion by adamoceallaigh** [PHP-CS-Fixer/PHP-CS-Fixer#7315] [Q&A] [answered]
On a project, I was using PHP-CS-Fixer v3.14.4. Currently, the project utilises PHP 8.0.

All internal symfony/* dependencies were using "php: >=8.0.2" while php-cs-fixer required "php: ^7.4 || ^8.0". 
Everything operated as desired.

[Symfony dependency example](https://packagist.org/packages/symfony/console)

When I updated to PHP-CS-Fixer to v3.27.0 to receive more improved rules. 
All internal symfony/* dependencies now state "php: >=8.1" while php-cs-fixer still requires "php: ^7.4 || ^8.0" according to docs. 
I think this is false and needs to be updated to "php: >=8.1" as composer check fails as dependencies are using PHP 8.1 and halts execution of my project.

This change wouldn't change anything in my regard as I will have to rollback the PR however subsequent people should be made aware of this fact that they require PHP version of 8.1 or higher to run PHP-CS-Fixer v3.27.0 ??

**Comment by Wirone** [comment, marked as answer]
I believe it's related to v3.25 where we dropped Doctrine dependencies, and it probably unlocked some updates on your side. It's not Fixer's responsibility to keep proper PHP requirement in your project. If your application is not compatible with PHP 8.1+ then you should set `8.0.*` constraint on your side and during Composer update Symfony packages' proper version will be matched. You can also set `platform.php` in you `composer.json`.

We can't change the constraint on our side to `>=8.1` because it would make Fixer unavailable for people with PHP 8.0.

**Reply by adamoceallaigh** [reply]
Thanks for your prompt reply, I agree and will take due actions
